apiserver/facades/agent/uniter/uniter.go

```diff
@@ -2333,18 +2325,18 @@ func (u *UniterAPI) goalStateRelations(
 
 		// Now gather the goal state.
 		for _, e := range endPoints {
-			var key string
-			app, err := u.st.Application(e.ApplicationName)
+			var appName string
+			appID, err := u.applicationService.GetApplicationIDByName(ctx, e.ApplicationName)
 			if err == nil {
-				key = app.Name()
-			} else if errors.Is(err, errors.NotFound) {
+				appName = e.ApplicationName
+			} else if errors.Is(err, applicationerrors.ApplicationNotFound) {
```

Comment:
Can you add a comment (or implement it), so the application name is retrieved once.

Reply:
Left a comment as such